File: spring-resilience4j/resilience4j-client/src/main/java/com/example/resilience4jclient/controller/HelloController.java
package com.example.resilience4jclient.controller;

import com.example.resilience4jclient.service.HelloService;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/")
public class HelloController {
  private HelloService helloService;

  public HelloController(HelloService helloService) {
    this.helloService = helloService;
  }

  @GetMapping
  public String index() {
    return helloService.getHello();
  }
}

File: spring-multi-project/web/src/main/kotlin/com/example/demo/controller/HelloController.kt
package com.example.demo.controller

import com.example.demo.service.HelloService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController("/")
class HelloController(
    private val helloService: HelloService
) {
    @GetMapping()
    fun getHello(): String {
        return helloService.say()
    }
}
